### PR TITLE
Remove legacy ProxyConnection stuff

### PR DESCRIPTION
**Pull request by gavinking**

### CODE REVIEW

**Comment by DavideD** [comment]
Did you also check if it's still used in Quarkus?

File: hibernate-reactive-core/src/main/java/org/hibernate/reactive/session/impl/ReactiveSessionImpl.java [hunk 1]
```diff
@@ -1721,7 +1721,7 @@ public ReactiveConnection getReactiveConnection() {
 
 	@Override
 	public void close() throws HibernateException {
-		throw new UnsupportedOperationException( "Non reactive close method called. Use 'reactiveClose()' instead." );
+		throw new UnsupportedOperationException( "Non-reactive 'close()' method called. Use 'reactiveClose()' instead." );
```
Comment (DavideD) [inline hibernate-reactive-core/src/main/java/org/hibernate/reactive/session/impl/ReactiveSessionImpl.java:1724]: Actually, we now have a specific error in the Log now: https://github.com/hibernate/hibernate-reactive/blob/main/hibernate-reactive-core/src/main/java/org/hibernate/reactive/logging/impl/Log.java#L243

**Comment by gavinking** [comment]
> Did you also check if it's still used in Quarkus?

Yes, @Sanne and I looked at this yesterday. Quarkus no longer uses it. `MutinyImplementor` *does* occur in one annotation, but that occurrence can just be deleted.

**Comment by DavideD** [comment]
Nice

File: hibernate-reactive-core/src/main/java/org/hibernate/reactive/session/impl/ReactiveSessionImpl.java (hunk 1 above)
Comment (gavinking) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/session/impl/ReactiveSessionImpl.java:1724]: OK, but we can fix that up in a subsequent commit.

**Pull request merged**